core/trino-main/src/main/java/io/trino/connector/CoordinatorDynamicCatalogManager.java

```diff
@@ -286,6 +289,93 @@ public void createCatalog(CatalogName catalogName, ConnectorName connectorName,
         }
     }
 
+    @Override
+    public void renameCatalog(CatalogName catalogName, CatalogName newCatalogName)
+    {
+        requireNonNull(catalogName, "catalogName is null");
+
+        catalogsUpdateLock.lock();
+        try {
+            checkState(state != State.STOPPED, "ConnectorManager is stopped");
+
+            if (!activeCatalogs.containsKey(catalogName)) {
+                throw new TrinoException(NOT_FOUND, format("Catalog '%s' does not exist", catalogName));
+            }
+            if (activeCatalogs.containsKey(newCatalogName)) {
+                throw new TrinoException(ALREADY_EXISTS, format("Catalog '%s' already exists", newCatalogName));
+            }
+
+            createCatalogLikeInternal(catalogName, newCatalogName, ImmutableMap.of());
+            catalogStore.removeCatalog(catalogName);
+            activeCatalogs.remove(catalogName);
+
+            log.info("Renamed catalog from: '%s' to: '%s'", catalogName, newCatalogName);
+        }
+        finally {
+            catalogsUpdateLock.unlock();
+        }
+    }
+
+    @Override
+    public void alterCatalog(CatalogName catalogName, Map<String, Optional<String>> properties)
+    {
+        requireNonNull(catalogName, "catalogName is null");
+
+        catalogsUpdateLock.lock();
+        try {
+            checkState(state != State.STOPPED, "ConnectorManager is stopped");
+
+            if (!activeCatalogs.containsKey(catalogName)) {
+                throw new TrinoException(NOT_FOUND, format("Catalog '%s' does not exist", catalogName));
+            }
+
+            createCatalogLikeInternal(catalogName, catalogName, properties);
+            log.info("Updated catalog: " + catalogName);
+        }
+        finally {
+            catalogsUpdateLock.unlock();
+        }
+    }
+
+    private CatalogConnector createCatalogLikeInternal(CatalogName oldCatalogName, CatalogName catalogName, Map<String, Optional<String>> properties)
```

Comment:
I find the name of this method confusing. Does it mean "create a catalog like 'internal'" or "the internal implementation of create-catalog-like"?